pytorch_lightning/plugins/native_amp.py

```diff
@@ -40,7 +42,7 @@ def backward(self, closure_loss, optimizer, opt_idx, *args, **kwargs):
         closure_loss = closure_loss.detach()
 
         # unscale gradient to allow analyze within `on_after_backward`
-        if not self.trainer.train_loop.should_accumulate() and self.trainer.train_loop.automatic_optimization:
+        if not self.trainer.train_loop.should_accumulate() and automatic_optimization:
```

Comment:
I don't think this is correct? We scale the loss regardless (line 30) and we have to unscale regardless of `automatic_optimization`. Is this unscaled elsewhere or something when `automatic_optimization=False`?

Reply:
cc @tchaton

Reply:
After offline conversation it seems like this is necessary, as the current behaviour is wrong. If the user calls `self.manual_backward` twice with the same optimizer, we call `scaler.unscale_(optimizer)` twice which throws an error. We only want to unscale gradients when we do the `step` calculation.

I suggest adding a function to the trainer such as `def manual_unscale_grads(optimizer)` which the user should call to unscale gradients before doing the optimizer step that handles unscaling of grads if using AMP. This kinda breaks the purpose of manual_backward handling scaling though, so open to suggestions.

Reply:
@SeanNaren I think we are not calling `scaler.update()` after every optimizer called `scaler.step()` or `optimizer.step()`. If `manual_optimization` is False, we have no control over what should be called after `optimizer.step()`, so would it be better provide `self.scaler` to end users?
I think this way makes sense for now b/c if they want to do their own optimization, they should also need to handle their own 16 bit for now. Like - we leave every 16 bit to end users except `with autocast():`
```py
loss = self.scaler.scale(loss)
self.manual_backward(loss)
self.scaler.unscale_(opt_a)
opt_a.step()
opt_a.zero_grad()
```

If we don't want users to handle 16 bit when `automatic_optimization=False`, we should rethink about calling `manual_backward` + `automatic_optimization` + 16 bit